pkgs/dart_tooling_mcp_server/lib/src/mixins/pub.dart

```diff
@@ -77,26 +77,27 @@ base mixin PubSupport on ToolsSupport, LoggingSupport
     description:
         'Runs a dart pub command for the given project roots, like `dart pub '
         'get` or `dart pub add`.',
-    inputSchema: ObjectSchema(
+    annotations: ToolAnnotations(title: 'pub', readOnlyHint: false),
```

Comment:
Note that we might want consider splitting up some of these tools into the (mostly) "read only" ones versus the ones that are not.

At least `pub get` should be considered read only probably, even though its technically not it only touches hidden files (other than the pubspec.lock, but it won't alter that if it exists already).